File: scripts/reset_password.js
#!/usr/bin/env node

/**
 * Script de Reset de Senha - Centelha Divina
 * 
 * Como usar:
 * node reset_password.js cinthiadutra@gmail.com nova_senha_123
 */

const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Cores para o terminal
const colors = {
  reset: '\x1b[0m',
  red: '\x1b[31m',
  green: '\x1b[32m',
  yellow: '\x1b[33m',
  blue: '\x1b[34m',
};

console.log(`${colors.blue}🔐 Script de Reset de Senha - Centelha Divina${colors.reset}\n`);

// Pegar argumentos da linha de comando
const email = process.argv[2];
const novaSenha = process.argv[3];

if (email && novaSenha) {
  resetPassword(email, novaSenha);
} else {
  // Modo interativo
  rl.question('Digite o email: ', (emailInput) => {
    rl.question('Digite a nova senha: ', (senhaInput) => {
      rl.close();
      resetPassword(emailInput.trim(), senhaInput.trim());
    });
  });
}

function resetPassword(email, senha) {
  console.log(`\n${colors.yellow}📝 Processando reset de senha...${colors.reset}`);
  console.log(`Email: ${email}`);
  
  if (senha.length < 6) {
    console.log(`${colors.red}❌ Erro: Senha deve ter no mínimo 6 caracteres!${colors.reset}`);
    process.exit(1);
  }

  // Instruções para executar no Supabase
  console.log(`\n${colors.blue}📋 Execute o seguinte SQL no Supabase Dashboard:${colors.reset}\n`);
  
  const sql = `UPDATE auth.users
SET 
  encrypted_password = crypt('${senha}', gen_salt('bf')),
  updated_at = NOW()
WHERE email = '${email}';`;

  console.log(colors.green + sql + colors.reset);
  
  console.log(`\n${colors.yellow}📍 Passos:${colors.reset}`);
  console.log('1. Acesse: https://supabase.com/dashboard');
  console.log('2. Selecione seu projeto');
  console.log('3. Vá em "SQL Editor"');
  console.log('4. Cole o comando SQL acima');
  console.log('5. Clique em "Run"');
  
  console.log(`\n${colors.green}✅ Depois você poderá fazer login com:${colors.reset}`);
  console.log(`   Email: ${email}`);
  console.log(`   Senha: ${senha}`);
  
  process.exit(0);
}

File: scripts/reset_password.dart
import 'dart:developer';
import 'dart:io';

import 'package:supabase_flutter/supabase_flutter.dart';

void main() async {
  log('🔐 Script de Reset de Senha - Centelha Divina\n');

  // Inicializar Supabase com service role key
  await Supabase.initialize(url: supabaseUrl, anonKey: supabaseServiceKey);

  final supabase = Supabase.instance.client;

  // Solicitar email
  stdout.write('Digite o email do usuário: ');
  final email = stdin.readLineSync()?.trim();

  if (email == null || email.isEmpty) {
    log('❌ Email inválido!');
    exit(1);
  }

  // Solicitar nova senha
  stdout.write('Digite a nova senha (mínimo 6 caracteres): ');
  final novaSenha = stdin.readLineSync()?.trim();

  if (novaSenha == null || novaSenha.length < 6) {
    log('❌ Senha deve ter no mínimo 6 caracteres!');
    exit(1);
  }

  // Confirmar senha
  stdout.write('Confirme a nova senha: ');
  final confirmaSenha = stdin.readLineSync()?.trim();

  if (novaSenha != confirmaSenha) {
    log('❌ As senhas não coincidem!');
    exit(1);
  }

  log('\n📝 Processando...');

  try {
    // Buscar usuário no Supabase Auth
    // Nota: Esta operação requer service_role key
    await supabase.auth.admin.updateUserById(
      email, // Na verdade precisamos do UUID, mas vamos tentar reset por email
      attributes: AdminUserAttributes(password: novaSenha),
    );

    log('✅ Senha redefinida com sucesso para $email!');
    log('');
    log('Agora você pode fazer login com:');
    log('  Email: $email');
    log('  Senha: $novaSenha');
  } catch (e) {
    log('❌ Erro ao redefinir senha: $e');
    log('');
    log('💡 Dica: Certifique-se de usar a service_role key do Supabase.');
    exit(1);
  }

  exit(0);
}

const String supabaseServiceKey =
    'SUA_SERVICE_KEY'; // Use a service_role key, não a anon key

// Configure suas credenciais do Supabase aqui
const String supabaseUrl = 'SUA_URL_SUPABASE';
